files/image_config/monit/container_checker

```diff
@@ -51,14 +51,20 @@ def get_expected_running_containers():
                     num_asics = multi_asic.get_num_asics()
                     for asic_id in range(num_asics):
                         expected_running_containers.add(container_name + str(asic_id))
-            elif feature_table[container_name]["state"] == 'always_enabled':
-                always_running_containers.add(container_name)
             else:
                 expected_running_containers.add(container_name)
-
+        if feature_table[container_name]["state"] == 'always_enabled':
+            if multi_asic.is_multi_asic():
```

Comment:
has_global_scope -- Is this true only for Database container?

Reply:
has_global_scope is True for other containers e.g. lldp, dhcp_relay, mux, etc
but here the if statement is under "if feature_table[container_name]["state"] == 'always_enabled'", which is only True for mux and databse, while mux doesn't seem to be a container as shown in 'docker ps -a'

admin@str-n3164-acs-2:~$ show feature status
Feature     State            AutoRestart     SystemState    UpdateTime           ContainerId    Version      SetOwner    CurrentOwner    RemoteState
----------  ---------------  --------------  -------------  -------------------  -------------  -----------  ----------  --------------  -------------
acms        enabled          enabled         up             2022-02-22 19:20:47  acms                        local       local           none
bgp         enabled          enabled                                                                         local
database    always_enabled   always_enabled                                                                  local
dhcp_relay  enabled          enabled         up             2022-02-22 19:21:10  dhcp_relay     20201231.54  kube        local
lldp        enabled          enabled         up             2022-02-22 19:20:53  lldp           20201231.54  kube        local
mux         always_disabled  enabled                                                                         local
pmon        enabled          enabled         up             2022-02-22 19:20:48  pmon           20201231.54  kube        local
radv        enabled          enabled         up             2022-02-22 19:21:06  radv           20201231.54  kube        local
restapi     enabled          enabled         up             2022-02-22 19:20:47  restapi                     local       local           none
snmp        enabled          enabled         up             2022-02-22 19:23:56  snmp           20201231.54  kube        local
swss        enabled          enabled                                                                         local
syncd       enabled          enabled                                                                         local
teamd       enabled          enabled                                                                         local
telemetry   enabled          enabled         up             2022-02-22 19:23:59  telemetry      20201231.54  kube        local

Reply:
The previous code on line 54 and 55 will add container with 'state' as 'always_enabled' to always_running_containers only if single-asic. I moved that condition one upper level, and let it decide separately for multi-asic and single-asic.